caom2-soda-server/src/main/java/ca/nrc/cadc/caom2/soda/SodaJobRunner.java

```diff
@@ -195,7 +195,7 @@ public URL toURL(int serialNum, URI uri, Cutout<Shape> pos, Cutout<Interval> ban
                     // get the optional label parameter value
                 	List<String> labels = customParams.get(PARAM_LABEL);
                 	String label = null;
-                	if (labels.size() > 0) {
+                	if (labels != null && labels.size() > 0) {
```

Comment:
check should be: `labels != null && !labels.isEmpty()` ... when you have a List and don't know the impl, asking for size() can be costly (eg LinkedList may have to count the items)

Reply:
Right. Thank you. Updated.